source/guide_buildbot.rst

```diff
@@ -0,0 +1,266 @@
+
+
+Configuration of the SSH connection
+===================================
+
+Step 1
+------
+
+In order to view the Buildbot master's webinterface, we need to forward the respective port through our SSH connection.
+
+You can either do this via the ``ssh`` command like so:
+
+::
+
+ [isabell@desktop ~] ssh -L 12345:localhost:12345 isabell@stardust.uberspace.de
+
+Or you can adjust your local ``~/.ssh/config`` file by adding the ``LocalForward`` option to the Uberspace host. The host entry would look something like this:
+
+::
+
+ Host stardust
+ 	HostName stardust.uberspace.de
+ 	User isabell
+ 	LocalForward 12345 localhost:12345
+
+You can then connect via
+
+::
+
+ [isabell@desktop ~] ssh stardust
+
+
+Step 2
+------
+
+Now that the connection is established with port forwarding, you can call up ``http://localhost:12345/`` in your browser to access the Buildbot_ webinterface! We have now basically completed the `'First Run' tutorial of the official manual <https://docs.buildbot.net/latest/tutorial/firstrun.html>`_ and you should be able to force an execution of the ``runtests`` builder.
+
+Integration with gitea
+======================
+
+One useful thing to do with BuildBot_ is to use it as a continuous integration runner. Since gitea_ also works on Uberspace but doesn't support 'direct' CI/CD integration like github and gitlab, we can use gitea_'s web hooks to trigger our BuildBot_ installation to do something.
+
+Step 1
+------
+
+For this, we will need to install the ``buildbot_gitea`` plugin for BuildBot, developed by Marvin Pohl of lab132. First, clone their git repository and then run the installation:
+
+::
+
+ [isabell@stardust ~] git clone https://github.com/lab132/buildbot-gitea.git
+ [isabell@stardust ~] cd buildbot-gitea
+ [isabell@stardust buildbot-gitea] pip install . --user
+
+Step 2
+------
+
+Now that we installed the ``buildbot_gitea`` plugin, we can use ``gitea`` as a dialect for accepting incoming webhook messages via ``http://localhost:12345/change_hook/gitea``. For this, return to editing the ``master.cfg`` from earlier. There, add the following to enable incoming webhook messages from gitea:
+
+::
+
+ c['www']['change_hook_dialects'] = {
+	'gitea': {
+		'secret': 'SomeSecretPassPhraseToAuthenticateGitea',
+	}}
+
+That's it! Restart your BuildBot_ master via ``buildbot restart master`` and continue by adding the webhook to the desired gitea repository!
+
+Step 3
+------
+
+Adding the webhook to a repository works pretty much as expected. Go to the desired repository, click on ``Settings > Webhooks > Add Webhook > Gitea`` and enter ``http://localhost:12345/change_hook/gitea`` as the URL and whatever you entered as a secret as secret. This of course assumes that you installed gitea on the same Uberspace host as BuildBot_.
```

Comment:
> This of course assumes that you installed gitea on the same Uberspace host as BuildBot.

Because of the mentioned web backend changes, this won't be possible in the future. Buildbot and gitlab need to be installed on the same uberspace, use SSH tunnels or expose buildbot via HTTPS.